src/main/java/com/cst438/controller/EnrollmentController.java

```diff
@@ -37,6 +87,17 @@ public void updateEnrollmentGrade(@RequestBody List<EnrollmentDTO> dlist) {
         //  find the Enrollment entity using enrollmentId
         //  update the grade and save back to database
 
+        try {
+            for (EnrollmentDTO dto : dlist) {
```

Comment:
Very clean. I'll take note of this format the implementation is robust. Future ill will try this with my code.